datafusion/optimizer/src/push_down_filter.rs

```diff
@@ -435,22 +436,37 @@ fn push_down_all_join(
             {
                 right_push.push(on)
             } else {
-                join_conditions.push(on)
+                on_filter_join_conditions.push(on)
             }
         }
     }
 
     // Extract from OR clause, generate new predicates for both side of join if possible.
     // We only track the unpushable predicates above.
-    if on_left_preserved {
+    if left_preserved {
         left_push.extend(extract_or_clauses_for_join(&keep_predicates, left_schema));
         left_push.extend(extract_or_clauses_for_join(&join_conditions, left_schema));
     }
-    if on_right_preserved {
+    if left_preserved {
```

Comment:
I thınk you meant `right_preserved` here.

Reply:
Oops